### DISCUSSION TITLE
如何使用mirai2.15扫码登录的新功能

### DISCUSSION THREAD

**Discussion by yujumpjump** [simple-robot/Discussions#113] [Q&A]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

大佬。现在mirai好多登录方式都挂掉了 我看2.15出现了扫码登录的方式。我通过maven导了2.15版本的。它说现在只支持手表协议和苹果协议使用扫码登录  我通过修改了json配置文件里面的协议 然后登录发现还是没有出现扫码登录都方式。请问怎么才能在simbot3里面使用mirai2.15的新功能呢 是否需要我自定义登录方式呢

**Comment by ForteScarlet** [comment]
mirai的 [v2.15.0-M1](https://github.com/mamoe/mirai/releases/tag/v2.15.0-M1) 中提到了
> 新的登录方法通过 `BotAuthorization` & `BotFactory.newBot(id: Long, authorization: BotAuthorization)` 登录

而不是使用原本的配置方式，因此暂时无法通过组件直接使用。需等待 mirai 更新 `v2.15.0` 的正式版后（及组件进行针对性更新后）才可使用。未来也许会考虑增加一些能用于兼容此情此景的方式，但是目前没有。
